ts_algorithms/fdk.py

```diff
@@ -0,0 +1,79 @@
+import numpy as np
+import torch
+import tomosipo as ts
+from .fbp import fbp
+
+
+def fdk_weigh_projections(op, projections, recalculate_weights):
+    dev = projections.device
+    u_num_pixels = op.range_shape[2]
+    u_space = (torch.arange(u_num_pixels, dtype=torch.float, device=dev)+0.5-(u_num_pixels/2))
+    v_num_pixels = op.range_shape[0]
+    v_space = (torch.arange(v_num_pixels, dtype=torch.float, device=dev)+0.5-(v_num_pixels/2))
+
+    result = torch.empty(projections.shape, dtype=torch.float, device=dev)
+
+    for i in range(op.range_shape[1]):
+        if recalculate_weights or i == 0:
+            src_to_det = torch.from_numpy(op.range.det_pos[i] - op.range.src_pos[i]).float().to(dev)
+            src_to_obj = torch.from_numpy(-op.range.src_pos[i]).float().to(dev)
```

Comment:
The main reason why I converted everything to 32bit float is so that you could do these computations on the GPU, this is not possible with 64 bit floats. You could of course also calculate the weights mat on the CPU and transfer it to the used device afterwards. I don't expect there to be much of a speed or accuracy difference, but I guess it would be better to err on the side of caution and do the calculations as 64 bits on the cpu. What do you think?